### ISSUE TITLE
[question] Use of evaluate_policy and early reset with DummyVecEnv (normalization)

### ISSUE THREAD

**Issue by caburu** [hill-a/stable-baselines#906]
I'm using `evaluate_policy` helper to compare PPO2 agent with my baselines. The baselines are not RL algorithms, but simple heuristics instead. I've implemented them following the OpenAI Gym interface in order to use the same `evaluate_policy` function.

My custom env generates random values on each `reset` call. So, I thought that using the same seed I was comparing exactly the same env behavior for PPO2 and the baselines.

But I've found that as I'm using normalization with PPO2 and `DummyVecEnv` do early resets at the end of an episode, in fact `reset` method is called twice per episode when I use `evaluate_policy`. On the other hand as I'm using the custom env directly with my baselines, `reset` method is called only once per episode using the same function. And, in this way, I'm not doing a fair comparison.

If you think it's worth changing `evaluate_policy` because of this, I could open a PR about it.

The idea would be call `reset` directly only once and after using while statement to count the number of episodes. Below is a code snippet with the suggestion.

```python
    if isinstance(env, VecEnv):
        assert env.num_envs == 1, "You must pass only one environment when using this function"        
        wrap_action = False
    else:
        env = DummyVecEnv([lambda:env])
        wrap_action = True


    episode_rewards, episode_lengths = [], []
    obs = env.reset()
    num_episodes = 0
    while num_episodes < n_eval_episodes:
        done, state = False, None
        episode_reward = 0.0
        episode_length = 0    
        while not done:
            action, state = model.predict(obs, state=state, deterministic=deterministic)
            if wrap_action: action = [action]
            obs, reward, done, _info = env.step(action)
            episode_reward += reward
            if callback is not None:
                callback(locals(), globals())
            episode_length += 1
            if render:
                env.render()
        num_episodes += 1
        episode_rewards.append(episode_reward)
        episode_lengths.append(episode_length)
    mean_reward = np.mean(episode_rewards)
    std_reward = np.std(episode_rewards)
```

**Comment by araffin** [comment]
>reset method is called twice per episode

Duplicate of https://github.com/DLR-RM/stable-baselines3/issues/60

The root cause is the `VecEnv`, we would appreciate a PR that solves the issue in both repos ;) (see comments for the solution)

**Comment by caburu** [comment]
I checked the comments there. For me it's not clear yet how `need_reset` would work. @Miffyli points out this:

    need_reset() function (returns True if at least one environment requires reset?).

**Comment by araffin** [comment]
I would say that `need_reset()` should return false until `.reset()` has been called once.
So the comment would be more: "returns True if the reset() method was never called"
